Nanotrasen Deserter Vault Ruin

Latentish opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
Adds a new ruin spawn to the icemoon called the "Nanotrasen Corporate Security Solutions Vault Gamma". The level is full of Nanotrasen deserters guarding a very secure vault. The level takes into account players who might want to plan a heist rather than run through the level in a normal manner, beefing up defenses in areas one might try to break in through. It would take about the same amount of effort to plan and mount a heist on the vault as it would to run through it plain and simple. The level is possible to complete with no pre-brought medical supplies, running as the Hyena captain alone, so it should be fine for other players to complete as well. The total pool of loot that can be gotten is as follows:
Weapons and Armor: Particle Acceleration Rifle, NT Saber SMG with firing pin, combat shotgun, fireaxe, medibeam, disablers, Runite Armor (platemail armor), 2 spare clips for the smg, upgraded weapon cell for the particle acceleration rifle, and a variety total of 16 loose shotgun shells.
Materials: 10 Capitalisium, 10 runite, 2 adamantine, 15 bananium, 20 gold, 5 bluespace polycrystals, 20 plasma, 20 uranium, 2000 credits.
Machines/vendors: Empty AI core, soda dispenser, booze dispenser, boozeomat, misc civilian vendors, sectech, autolathe.
Equipment: Several basic toolsets, filled surgery duffelbag, stethoscope, medkit, rolling table, welding skull gasmask, insulated gloves, CNS rebooter.
Misc: Security weapon voucher, red phone, top secret documents (folder), hand tele, chaos donut, Oingo Boingo punch-face, positronic brain, MMI, HONK central control module, posters, bedsheets, sentience potion.

![image](https://user-images.githubusercontent.com/55075747/151680681-102bb081-6ae5-413f-bacb-195c86ade945.png)
![image](https://user-images.githubusercontent.com/55075747/151680660-eb0e9890-6afd-41b0-a23c-cc9a1355c7bd.png)
![image](https://user-images.githubusercontent.com/55075747/151680666-70b01040-6c50-4548-af8e-5ba2689084ca.png)
![image](https://user-images.githubusercontent.com/55075747/151680673-c5fba3a9-1f87-4c45-9152-d0f11331bf52.png)

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
The Icemoon is rather lacking in number of ruins when compared to other planets. This level adds some higher level loot to the game, while fitting the niche for a more difficult challenge. 
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog
:cl:Phoaly
add: Added a new map "maps\randomruins\iceruins\icemoon_surface_corporate_rejects.dmm"
code: Added a datum and several areas for the map in "whitesands\code\datums\ruins\icemoon.dm" and "code\game\area\areas\ruins\icemoon.dm" respectively.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Latentish committed:
Nanotrasen Deserter Vault Ruin

A ruin based around a Nanotrasen ultra secure vault. It should spawn on the ice planet. This commit adds everything.

Latentish commented:
Feedback on the loot pool would be appreciated.

UltraNumeron commented:
Seems pretty good honestly only issue i have is that for the loot pool it seems a bit too easy if im being honest, could maybe add some turrets on the outside if you can get those to work with maybe some enemies inside the vault itself

Firefox13 commented:
> Feedback on the loot pool would be appreciated.

R&D documents

Latentish commented:
> 

I don't really understand what that means.

UltraNumeron commented:
> > 
> 
> I don't really understand what that means.

They likely mean papers containing rnd points, like the discoveries you can get from  dissecting corpses and then slap in a research console for points.

Latentish commented:
> > > 
> > 
> > 
> > I don't really understand what that means.
> 
> They likely mean papers containing rnd points, like the discoveries you can get from dissecting corpses and then slap in a research console for points.

Makes sense, I'll swap those out for something else.

1glitchycent commented:
replace the sniper rifle with a tactical or antique Egun, NT doesn't into ballistics

Latentish commented:
> replace the sniper rifle with a tactical or antique Egun, NT doesn't into ballistics

I mean, they have their own brand of SMG. But people were having an issue with the sniper rifle anyway, so I'll find another funny gun. Maybe a pulse rifle?

Apogee-dev commented:
NT definitely does into ballistics, but the sniper rifle isn't one of theirs and is seriously busted. Make it a captain's laser gun instead, that's high quality loot without being outright broken.

Latentish committed:
Map Changes

-Replaces Glockroach family with Jack and Jill, Slaughter and Laughter demons.
-Replaces Sniper Rifle with Particle Acceleration Rifle.
-Replaces Sniper Rifle ammo with a single upgraded weapon power cell.
-Adds a sentience potion and cns rebooter implant to vault safe since there were only mats in it otherwise.

Latentish committed:
Minor commit

Removes a cybernetic that should have been removed before the last commit.

Latentish commented:
What are run linters?

ZephyrTFA commented on code/game/area/areas/ruins/icemoon.dm:
```diff
@@ -38,3 +38,16 @@
+	name = "NT Security Solutions Site Gamma Engineering"
+	icon_state = "dk_yellow"
+
+	/area/ruin/unpowered/corprejectvault
```
```suggestion
/area/ruin/unpowered/corprejectvault
```

Kapu1178 commented:
> What are run linters?

linters are the per-compilation scan of the code to find things that are wrong with it.

Latentish committed:
Update code/game/area/areas/ruins/icemoon.dm

I'm dumb and didn't realize I did this. Also didn't realize linters was the code checker, so I didn't realize to check the code. I know now! Thanks for the help.

Co-authored-by: Zephyr <12817816+ZephyrTFA@users.noreply.github.com>

Latentish committed:
Merge branch 'shiptest-ss13:master' into NT-Ice-Vault

Latentish committed:
Adds the knight gear design disk.

Adds the "magical disk of smithing" to the safe.

Latentish commented:
Mergema and/or reviewma.

triplezeta commented on whitesands/code/datums/ruins/icemoon.dm:
```diff
@@ -34,3 +34,9 @@
+	name = "NT Security Solutions Site Gamma"
+	id = "corpreject"
+	description = "Nanotrasen Corporate Security Solutions vault site Gamma."
+	suffix = "icemoon_surface_corporate_rejects.dmm"
```
Move to the unmodular ruin datums file please

Latentish replied:
Donezo!

triplezeta reviewed (changes requested):
Solid looking ruin overall, just need to address this modularization bit

triplezeta commented:
Balance issues with the ruin can be addressed after a few rounds of live tests

Latentish committed:
Unmodularizes your Modularization

Moves the datum to an unmodularized folder.

ZephyrTFA reviewed (approved):
I don't really approve maps, but we also don't have a dedicated maptainer so I'll toss my approval on this

Latentish commented:
> I don't really approve maps, but we also don't have a dedicated maptainer so I'll toss my approval on this

Glad you like it at the least! =)(
